### PR TITLE
Setup age and sex standardisation

### PR DESCRIPTION
**Pull request by chrisjwood16**
1) Script to regroup AgeGroup - lowest level 18-24 - (study def only includes >=18yrs). 
And regroup age_band - 0-19 join with 20-29 as 18-29 (study def only includes >=18yrs). 

2) Update rate_calculations_demographics.py to standardise for age and sex. Or age standardise for sex measure and sex standardise for age measure. Include population measure standardisation.

### CODE REVIEW

**Comment by chrisjwood16** [comment]
Think it probably should have testing setup for the standardisation apply functions.
From a look at the processed dummy data mostly seems reasonable although I think needs a closer look at the age_band and ethnicity measure as results possibly slightly odd.

File: analysis/rate_calculations_demographics.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+def standardise_rates_sex_apply(by_sex_row, standard_pop):
+    row_sex_group = by_sex_row['sex']
+    pop_ratio = standard_pop.loc[(standard_pop["sex"]==row_sex_group), "uk_pop_ratio"]
+    if row_sex_group == 'missing' or pop_ratio.empty:
+        row_standardised_rate = np.nan
+    else:
+        pop_ratio = pop_ratio.values[0]
+        row_standardised_rate = by_sex_row['agesex_rates'] * pop_ratio
+    return row_standardised_rate
```
Comment (LFISHER7) [inline analysis/rate_calculations_demographics.py:85]: These could probably be combined into a single function

File: analysis/tests/test_correct_age_groups.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            "AgeGroup": pd.Series(["18-24", "18-24", "18-24", "18-24", "40-44", "40-44", "40-44", "40-44", "18-24", "18-24", "18-24", "18-24", "40-44", "40-44", "40-44", "40-44"]),
+			"sex": pd.Series(['F', 'F', 'M', 'M', 'F', 'F', 'M', 'M', 'F', 'F', 'M', 'M', 'F', 'F', 'M', 'M']),
+			"demographic": pd.Series([0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1]),
+			"date": pd.Series(["01/04/2019", "01/04/2019", "01/04/2019", "01/04/2019",  "01/04/2019", "01/04/2019", "01/04/2019", "01/04/2019", "01/05/2019", "01/05/2019", "01/05/2019", "01/05/2019", "01/05/2019", "01/05/2019", "01/05/2019", "01/05/2019"]),       
+            "numerator": pd.Series([25, 45, 65, 85, 1, 2, 3, 4, 105, 125, 145, 165, 5, 6, 7, 8]),
+            "population": pd.Series([246, 258, 324, 244, 145, 167, 147, 178,  315, 300, 339, 383, 136, 137, 157, 147]),
+            "value": pd.Series([0.1016260162601626, 0.1744186046511628, 0.2006172839506173, 0.3483606557377049, 0.0068965517241379, 0.0119760479041916, 0.0204081632653061, 0.0224719101123596, 0.3333333333333333, 0.4166666666666667, 0.4277286135693215, 0.4308093994778068, 0.0367647058823529, 0.0437956204379562, 0.0445859872611465, 0.054421768707483])
```
Comment (LFISHER7) [inline analysis/tests/test_correct_age_groups.py:49]: You could calculate this from the other columns here to save hardcoding these values

**Review by LFISHER7 — APPROVED** [review]
This all looks to make sense to me. I think this could be refactored slightly to make the steps a bit clearer, which would then help with deciding what other tests to add. You could run this for now and we can return to it later, some of it may be covered in the work of turning this into a reusable action.

**Pull request merged**